uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/filters/date-type.js

```diff
@@ -1,38 +1,35 @@
 define([
-    'tasks-assignments/filters/filters'
-], function (filters) {
+    'tasks-assignments/filters/filters',
+    'common/lodash'
+], function (filters, _) {
     'use strict';
 
     filters.filter('filterByDateType',['$filter', '$rootScope', '$log', function ($filter, $rootScope, $log) {
         $log.debug('Filter: filterBy.dateType');
 
-        return function(inputArr, dateTypeArr) {
-            var filteredArr = [],
-                i = 0,
-                ii,
-                inputArrLen = inputArr.length,
-                dateTypeArrLen = dateTypeArr.length,
-                dateContactList,
-                dateContactListLen;
-
-            if (!inputArrLen || !dateTypeArrLen) {
-                return inputArr;
+        return function(dateContactList, dateTypeList) {
+            var filteredDateContactList = [];
+
+            if (!dateContactList.length || !dateTypeList.length) {
+                return dateContactList;
             }
 
-            for (i; i < inputArrLen; i++) {
-                ii = 0,
-                dateContactList = inputArr[i].dateContactList,
-                dateContactListLen = dateContactList.length;
+            _.forEach(dateContactList, function (dateContact) {
+              var dateContactList = dateContact.dateContactList;
 
-                for (ii; ii < dateContactListLen; ii++) {
-                    if (dateTypeArr.indexOf(dateContactList[ii].type) !== -1) {
-                        filteredArr.push(inputArr[i]);
-                        break;
-                    }
+              dateContactList && _.forEach(dateContactList, function (singleDateContact) {
```

Comment:
I think this is an example of overusing `() && ()`. It is really confusing to read something like that. Let's stick with classic `if (...) { _.foreach... }`